Arm backend: Add 16A8W support and test for slice operation

Ninja91 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13802
* #13801
* #13800
* #13799
* __->__ #13798
* #13797
* #13796

Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218

Ninja91 committed:
Add 16A8W support and test for slice operation

Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

[ghstack-poisoned]

digantdesai commented on backends/arm/test/ops/test_slice.py:
```diff
@@ -119,3 +126,102 @@ def test_slice_tensor_vgf_INT(test_data: torch.Tensor):
     pipeline.run()
+
+
+def get_symmetric_a16w8_slice_quantizer(per_channel_quantization=False):
```
same code duplication issue here

Ninja91 replied:
Will dedupe once all these PRs are pushed. For now, let's keep them independent.

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for slice operation"


Add 16A8W quantization support and test for the slice operation in ExecutorTorch ARM backend.

This follows the pattern established for linear, mul, sigmoid, and tanh operations, extending int16 support to slice operations.

Changes:
- Add INT16 dtype validation support in op_slice.py
- Add test_slice_tensor_16a8w_tosa_INT test function
- Enable test_slice.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80511095](https://our.internmc.facebook.com/intern/diff/D80511095/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]
